setup.py

```diff
@@ -1,12 +1,48 @@
 #!/usr/bin/env python
 # vim: set ts=2 sw=2 tw=99 et: 
+import multiprocessing as mp
+import os
 import sys
 from setuptools import setup, find_packages
 
 try:
   import sqlite3
 except:
   raise SystemError('py-sqlite3 must be installed')
+
+def child_main():
+  import sys
+  sys.path.pop(0)
+  try:
+    import ambuild2.util
+  except:
+    sys.exit(2)
+  if getattr(ambuild2.util, 'INSTALLED_BY_PIP_OR_SETUPTOOLS', False):

```

Comment:
I think a sentinel object would work better, that way any arbitrary value can be detected and fail the install.
```python
sentinel = object() # Plain object for minimal memory usage.
if getattr(ambuild2.util, 'INSTALLED_BY_PIP_OR_SETUPTOOLS', sentinel) is not sentinel:
```

Reply:
Sanity check:
Python 2.7+ confirms "object()" results in new objects, not a cached object.

Reply:
I'm trying to keep this simple - Python is... not my first choice anymore, and maintaining AMBuild is hard. I have to remember a lot of quirks. It should have been written in Go, but that wasn't around when AMBuild was needed. Anyway, "True" is a lot more clear than "object()", especially given the boolean phrasing of the variable name.